### DISCUSSION TITLE
Touch with a Waveshare 7"HDMI

### DISCUSSION THREAD

**Discussion by SoenkeRapp** [rsta2/circle#462] [Q&A]
A while ago I set up a simple circle project to display graphics from a Raspberry0 on this display("7inch HDMI LCD (H)"with USB), which workes fine.
Now I wanted to add some input functionality with the touchscreen, but can't even make the touchscreen sample/calibrator run correctly. The start looks ok to me, but there is no event fired when touching.

![20240724_124840](https://github.com/user-attachments/assets/a46bbd2c-bded-45c7-9608-6b881a6f4a06)

When using a debian image on the same setup, the Touch works, so hardware should be ok.
Maybe there are simply some configurations that I missed, but I don't know which and have no real idea, where to start to investigate deeper. It would be great, if somebody could point me into the right direction,.

**Comment by rsta2** [comment]
I would like to have a look at the HID report descriptor of your touchscreen. Can you please install the *kernel.img* in from the attached .zip file, start your RPi and take and send a photo from the larger hexdump, which should appear. Then, when the program runs and you touch the screen, it should show smaller hexdumps, which are the HID reports from the touch screen. Will see, what happens.

[kernel.zip](https://github.com/user-attachments/files/16361810/kernel.zip)

**Reply by SoenkeRapp** [reply]
Thank you for addressing my problem (and Thank You for this great environment anyway).
The output doesn't fit completely on screen in standard font, but the dump section is this:
![dump](https://github.com/user-attachments/assets/a396e79d-abae-441c-af82-5cad4b367ba8)

**Reply by rsta2** [reply]
You are welcome and thanks for the photo! Unfortunately I haven't expected such a big HID report descriptor and my OCR program fails to scan it. To be able to test the Circle HID parser with it, we must do another test with the attached kernel image.

The procedure is the same as before, except that the program runs only 20 seconds and then saves the log into the file *CIRCLE.LOG* on the SD card. So please install the kernel image on the SD card, start the system and after the "Just use your touchscreen!" message press a few points on the screen (not too many). The program tells you, when you can power off the system. Then take the SD card, copy the file *CIRCLE.LOG* and send it here. Thanks.

[kernel.zip](https://github.com/user-attachments/files/16375918/kernel.zip)

**Comment by SoenkeRapp** [comment]
That looked promising :-)
I touched the screen 5 times and got the attached output.
[CIRCLE.LOG](https://github.com/user-attachments/files/16388204/CIRCLE.LOG)

**Reply by rsta2** [reply]
Thanks for the log! I analyzed the HID report descriptor in the log file and what Circle is doing with it, and everything looks all right. Also the received event reports should be decoded all right.

You wrote, that you got no event from the touch screen. With the latest kernel test image, one can see, there are touch screen events. This program has only two modifications, compared to the normal *sample/28-touchscreen*, beside saving the logging output, which should have no influence:

1. The initialization for the Raspberry Pi Official 7" touch screen has been removed (`m_RPiTouchScreen.Initialize()`). Because this initialization is done after the USB initialization (`m_USBHCI.Initialize()`, the Official 7" touch screen, gets the device name "touch2", when an USB touch screen is attached. An Official 7" touch screen may be detected, even when there is no such touchscreen attached, because it's up to the firmware, what it is doing here. If the Official 7" touch screen would receive the name "touch1", the normal sample program would work with it. But I cannot imagine, how this should happen.

2. The call to `pTouchScreen->SetCalibration()` has been removed. So if there would be an option `touchscreen=` in the file *cmdline.txt* on the SD card, it would be ignored in the test program, while the normal sample program could produce some unknown failure with an invalid `touchscreen=` setting. Do you have an option `touchscreen=` in the file *cmdline.txt*? If so, please remove this option and try the *sample/28-touchscreen* again.

If this still does not work, you could add the option `usbpowerdelay=1500` to the first line of the file *cmdline.txt* on the SD card. If this file does not exist, create one. This option increases the time, Circle waits for USB devices to come up. Some USB devices need this.

In any case, if your touch screen requires calibration - I don't know, if this is the case - you must run the touchscreen calibrator first and write a valid `touchscreen=` option (with 4 comma-separated parameters) to *cmdline.txt*. Otherwise the touchscreen will now work right. There is also some chance, that your touchscreen does not need calibration. In this case, the `touchscreen=` option must be removed from the file *cmdline.txt*.

**Comment by SoenkeRapp** [comment]
Thank you for investigating the logfile and the recommandations.
I tried them, but unfortunately to no avail. I then upgraded to version 47, and now everything is working as expected. Maybe there has been some improvement in the environment, or I mixed up my older version somewhere (the more likely variant...)
Thank you for your great support!

**Reply by rsta2** [reply]
Ah yes, it was this change in Circle 46: #418. You are welcome.
